modules/dag-l1/src/main/scala/org/tessellation/dag/l1/infrastructure/transaction/TransactionService.scala

```diff
@@ -0,0 +1,116 @@
+package org.tessellation.dag.l1.infrastructure.transaction
+
+import cats.data.{NonEmptyList, NonEmptySet}
+import cats.effect.{Ref, Sync}
+import cats.syntax.alternative._
+import cats.syntax.applicative._
+import cats.syntax.either._
+import cats.syntax.flatMap._
+import cats.syntax.functor._
+import cats.syntax.option._
+import cats.syntax.show._
+
+import scala.collection.immutable.SortedSet
+import scala.util.control.NoStackTrace
+
+import org.tessellation.dag.l1.domain.transaction.TransactionService
+import org.tessellation.dag.transaction.filter.Consecutive
+import org.tessellation.ext.collection.MapRefUtils._
+import org.tessellation.schema.address.Address
+import org.tessellation.schema.transaction.{Transaction, TransactionOrdinal, TransactionReference}
+import org.tessellation.security.Hashed
+import org.tessellation.security.signature.Signed
+
+import io.chrisdavenport.mapref.MapRef
+
+object TransactionService {
+
+  def make[F[_]: Sync]: F[TransactionService[F]] =
+    for {
+      lastAccepted <- MapRef.ofConcurrentHashMap[F, Address, TransactionReference]()
+      waitingTransactions <- Ref.of[F, Map[Address, NonEmptySet[Signed[Transaction]]]](Map.empty)
```

Comment:
isn't it a MapRef ?

Reply:
It isn't. I'm atomically pulling transactions from this cache and given that after a bit of thought decided that it's better to use `Ref[Map[]]` then `MapRef` here.